faros_event.sh

```diff
@@ -0,0 +1,470 @@
+    fi
+
+    if !(($dry_run))
+    then
+        emitToFaros
+    else
+        echo "Dry run: Event NOT emitted to Faros"
+    fi
+    exit 0
+}
+
+function parseArgs() {
+    # Loop through arguments and process them
+    while (($#));
+    do
+        case "$1" in
+            -k|--api_key)
+                faros_api_key="$2"
+                shift 2
+                ;;
+            -a|--application)
+                application_name="$2"
+                shift 2
+                ;;
+            -c|--commit)
+                commit_sha="$2"
+                shift 2 
+                ;;
+            -p|--pipeline)
+                pipeline_uid="$2"
+                shift 2
+                ;;
+            -e|--environment)
+                deployment_env="$2"
+                shift 2
+                ;;
+            -ds|--deploy_status)
+                deployment_status="$2"
+                shift 2
+                ;;
+            -bs|--build_status)
+                build_status="$2"
+                shift 2
+                ;;
+            --ci_org)
+                ci_org_uid="$2"
+                shift 2
+                ;;
+            --vcs_source)
+                vcs_source="$2"
+                shift 2
+                ;;
+            --vcs_org)
+                vcs_org_uid="$2"
+                shift 2
+                ;;
+            -r|--repo)
+                repository="$2"
+                shift 2
+                ;;
+            -g|--graph)
+                faros_graph="$2"
+                shift 2
+                ;;
+            --url)
+                faros_api_url="$2"
+                shift 2
+                ;;
+            --dry_run)
+                dry_run=1
+                shift
+                ;;
+            --print_event)
+                print_event=1
+                shift
+                ;;
+            --debug)
+                debug=1
+                shift
+                ;;
+            *)
+                echo "Unrecognized flag: $1"
+                exit 1
+        esac
+    done
+}
+
+function validateInput() {
+    # Required fields: If flag missing fall back to env var
+    faros_api_key=${faros_api_key:-$FAROS_API_KEY}
+    application_name=${application_name:-$APPLICATION_NAME}
+    commit_sha=${commit_sha:-$COMMIT_SHA}
+    pipeline_uid=${pipeline_uid:-$PIPELINE_UID}
+    ci_org_uid=${ci_org_uid:-$CI_ORG_UID}
+
+    # Optional fields: If flag missing fall back to env var then defualt
+    FAROS_GRAPH=${FAROS_GRAPH:-"default"}
+    faros_graph=${faros_graph:-$FAROS_GRAPH}
+
+    FAROS_API_URL=${FAROS_API_URL:-"https://prod.api.faros.ai"}
+    faros_api_url=${faros_api_url:-$FAROS_API_URL}
+
+    # Optional fields (No flag offered): If unset use default
+    ORIGIN=${ORIGIN:-"Faros_Script_Event"}
+    SOURCE=${SOURCE:-"Faros_Script"}
+    APPLICATION_PLATFORM=${APPLICATION_PLATFORM:-"NA"}
+    START_TIME=${START_TIME:-$( date +%s000000000 | cut -b1-13 )} # default now
```

Comment:
Does the date produce a correct result on all OS-es? You might want to start adding tests with GitHub actions and OS matrix. For example see - https://github.com/faros-ai/cli/blob/master/.github/workflows/ci.yml#L22

Reply:
You can probably use dry run mode for testing and assert the output with expected

Reply:
Good idea. I will see what I can do it the way of testing.